R/data-survey.R

```diff
@@ -269,3 +288,50 @@ calc_survey_hiv_indicators <- function(survey_meta,
 
   val
 }
+
+
+#' Find Calendar Quarter Midpoint of Two Dates
+#'
+#' @param start_date vector coercibel to Date
+#' @param end_date vector coercibel to Date
+#'
+#' @return A vector of calendar quarters
+#'
+#' @examples
+#' start <- c("2005-04-01", "2010-12-01", "2016-01-01")
+#' end <-c("2005-08-01", "2011-05-01", "2016-06-01")
+#'
+#' @export
+get_mid_calendar_quarter <- function(start_date, end_date) {
+
+  start_date <- lubridate::decimal_date(as.Date(start_date))
+  end_date <- lubridate::decimal_date(as.Date(end_date))
+
+  stopifnot(start_date <= end_date)
+  stopifnot(!is.na(start_date))
+  stopifnot(!is.na(end_date))
+  
+  date4 <- (start_date + end_date) / 2
+  date4 <- round(4 * date4) - 1
+
+  paste0("CY", date4 %/% 4, "Q", date4 %% 4 + 1)
+}
+
+
+#' Read Multiple Shape Files in ZIP Archive
+#'
+#' Reads all files in ZIP archive `zfile` matching `pattern` with
+#' function `read_fn` and returns as a list.
+#'
+#' @param zfile path to a zip directory
+#' @param pattern string pattern passed to [`list.files`].
+#' @param read_fn function used to read matched files.
+#'
+#' @export
+read_sf_zip_list <- function(zfile, pattern = "shp$", read_fn = sf::read_sf) {
```

Comment:
pattern would be slightly better as `\\.shp$`

Reply:
done in e17a9f3